core/agent.py

```
@@ -174,21 +193,70 @@ def run(
+                self.lifecycle.emit(
+                    MessageStartEvent(turn_number=self.lifecycle.get_turn_number())
+                )
                 try:
                     assistant_response = self.llm.chat(self.conversation)
+                    self.lifecycle.emit(
+                        MessageUpdateEvent(
+                            turn_number=self.lifecycle.get_turn_number(),
+                            delta=assistant_response,
+                        )
+                    )
                 except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)